### PR TITLE
feat: pin or unpin a post

### PR DESCRIPTION
**Pull request by sshanzel**
### Describe what this PR does
- Pin or unpin a post
- Refactor implementations to using `ConditionalWrapper` for flags.
- Created a reusable function to be utilize on the prop for ConditionalWrapper.

## Events

Did you introduce any new tracking events?
Don't forget to update the [Analytics Taxonomy sheet](https://docs.google.com/spreadsheets/d/18Lv7zXges9QfVX5VYL1a-Hyl0e1sQ3sLr0OK8YZWKXI/edit#gid=0)

Log the new/changed events below:

| Type   | event_name  | value |
|--------|-------------|-------|
| Change/New | event name  | extra: { ... } |

### On those affected packages:
- [ ] Have you done sanity checks in the webapp?
- [ ] Have you done sanity checks in the extension?
- [ ] Does this not break anything in companion?

### Did you test the modified components media queries?
- [ ] MobileL (420px)
- [ ] Tablet (656px)
- [ ] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

WT-1340 #done

### CODE REVIEW

File: packages/shared/src/components/PostOptionsMenu.tsx
Comment (sshanzel) [reply packages/shared/src/components/PostOptionsMenu.tsx:28]: I 100% agree with that. Let me create one for the icons folder.

**Review by TomDowling — APPROVED** [review]
Looks good for the icon stuff, but of course i'd wait for others approval fist still :)

File: packages/shared/src/components/PostOptionsMenu.tsx
Comment (rebelchris) [inline packages/shared/src/components/PostOptionsMenu.tsx:131]: Is this not something we can handle within the hook?
It seems like the wrong place to handle it per implementation.

File: packages/shared/src/graphql/feed.ts
Comment (rebelchris) [inline packages/shared/src/graphql/feed.ts:176]: Why do you add them conditional? I think we should just clean them into a FeedPost no?

File: packages/shared/src/hooks/usePostMenuActions.ts
Comment (rebelchris) [inline packages/shared/src/hooks/usePostMenuActions.ts:73]: I would handle the mutation success here actually, the hook seems like the perfect place to have this logic.

File: packages/shared/src/components/cards/PostList.tsx
Comment (rebelchris) [inline packages/shared/src/components/cards/PostList.tsx:46]: I'm not really sure this is the right way.
Multiple places doing the same thing, is there not a way we can uniform a cards raisedLabel based on props?

if it has `pinnedAt` show pinned, if it has trending -> show that, else nothing?
Keep in mind any post can be pinned.

**Review by rebelchris — CHANGES_REQUESTED** [review]
Looks great already.
Would love to see if we can make the card raised label a bit more generic for future proofing.

File: packages/shared/src/components/PostOptionsMenu.tsx
Comment (sshanzel) [reply packages/shared/src/components/PostOptionsMenu.tsx:131]: Have actually done that initially. But then I saw `PostOptionsMenu` handling all the toast and invalidations so I've thought of keeping everything in the same place. WDYT?

File: packages/shared/src/components/cards/PostList.tsx
Comment (sshanzel) [reply packages/shared/src/components/cards/PostList.tsx:46]: Let me check how we can uniform it.

**Comment by sshanzel** [comment]
@rebelchris have refactored them to use the same container wrapper. Feel free to test the pin/unpin function in our preview deployment.

https://preview.app.daily.dev/squads/dailydev2

**Review by rebelchris — CHANGES_REQUESTED** [review]
Minor remarks, almost there.

File: packages/shared/src/components/cards/FeedItemContainer.tsx
Comment (sshanzel) [reply packages/shared/src/components/cards/FeedItemContainer.tsx:23]: Since we only need this in one place now, there's no need to create a component. Hence, applied here manually as this will be the container of all Feed items.

Will remove unused components.

File: packages/shared/src/components/cards/FlaggedContainer.tsx
Comment (rebelchris) [inline packages/shared/src/components/cards/FlaggedContainer.tsx:20]: Where do you use this?
I think you can just delete this whole file actually.

**Review by rebelchris — CHANGES_REQUESTED** [review]

File: packages/shared/src/components/cards/FlaggedContainer.tsx
Comment (sshanzel) [reply packages/shared/src/components/cards/FlaggedContainer.tsx:20]: We are still using the `getFlaggedContainer`. Not anywhere atm but kept it there if anyone needs it. But if you want we can definitely remove this one.

File: packages/shared/src/components/cards/FlaggedContainer.tsx
Comment (sshanzel) [reply packages/shared/src/components/cards/FlaggedContainer.tsx:10]: This one is used.

**Review by rebelchris — APPROVED** [review]
Looks great!

**Pull request merged**
